Frameworks/Foundation/NSIndexSet.mm

```diff
@@ -350,72 +314,109 @@ - (BOOL)isEqualToIndexSet:(NSIndexSet*)otherSet {
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (NSUInteger)indexPassingTest:(BOOL (^)(NSUInteger, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self indexWithOptions:0 passingTest:predicate];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (NSIndexSet*)indexesPassingTest:(BOOL (^)(NSUInteger, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self indexesWithOptions:0 passingTest:predicate];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
-- (NSUInteger)indexWithOptions:(NSEnumerationOptions)opts passingTest:(BOOL (^)(NSUInteger, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSUInteger)indexWithOptions:(NSEnumerationOptions)options passingTest:(BOOL (^)(NSUInteger, BOOL*))predicate {
+    return [self indexInRange:sc_unlimitedRange options:options passingTest:predicate];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
-- (NSIndexSet*)indexesWithOptions:(NSEnumerationOptions)opts passingTest:(BOOL (^)(NSUInteger, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSIndexSet*)indexesWithOptions:(NSEnumerationOptions)options passingTest:(BOOL (^)(NSUInteger, BOOL*))predicate {
+    return [self indexesInRange:sc_unlimitedRange options:options passingTest:predicate];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
-- (NSUInteger)indexInRange:(NSRange)range options:(NSEnumerationOptions)opts passingTest:(BOOL (^)(NSUInteger, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSUInteger)indexInRange:(NSRange)range options:(NSEnumerationOptions)options passingTest:(BOOL (^)(NSUInteger, BOOL*))predicate {
+    NSIndexSet* set = [self indexesInRange:range options:options passingTest:predicate];
+    return (options & NSEnumerationReverse) ? [set lastIndex] : [set firstIndex];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
-- (NSIndexSet*)indexesInRange:(NSRange)range options:(NSEnumerationOptions)opts passingTest:(BOOL (^)(NSUInteger, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSIndexSet*)indexesInRange:(NSRange)range options:(NSEnumerationOptions)options passingTest:(BOOL (^)(NSUInteger, BOOL*))predicate {
+    __block NSMutableIndexSet* ret = [NSMutableIndexSet indexSet];
+    [self enumerateIndexesInRange:range
+                          options:options
+                       usingBlock:^(NSUInteger index, BOOL* stop) {
+                           if (predicate(index, stop)) {
+                               [ret addIndex:index];
+                           }
+                       }];
+
+    return ret;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
-- (void)enumerateRangesInRange:(NSRange)range options:(NSEnumerationOptions)opts usingBlock:(void (^)(NSRange, BOOL*))block {
-    UNIMPLEMENTED();
+- (void)enumerateRangesUsingBlock:(void (^)(NSRange, BOOL*))block {
+    [self enumerateRangesWithOptions:0 usingBlock:block];
```

Comment:
please always call the most-specific version in these shim implementations; you'll avoid an entire level of dynamic dispatch and a stack frame.